### ISSUE TITLE
Dagger Hilt Gradle Plugin will not perform ASM bytecode transformation when building with AGP 7.1.0-rc01

### ISSUE THREAD

**Issue by jilulu** [google/dagger#3149]
## Dependencies and versions: 

| Dependency | Version | 
| --- | --- |
|Android Studio | Bumblebee 2021.1.1 RC 1 |
|Kotlin Version | 1.6.10 |
|Dagger Hilt | 2.40.5 |
|Robolectric | 4.6 |

## Scenario

When requesting an `ActivityScenario` in Robolectric unit tests, the `Activity` being created was not being processed by Hilt Gradle Plugin, resulting in dependencies in the activity not being satisfied by Hilt dependency injection. However, when building and running the same app, everything works fine, and dependencies are being satisfied. 

That is, the activity below cannot be launched in tests: 

```kotlin 
@AndroidEntryPoint
class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var criticalDependency: CriticalDependency

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (criticalDependency == null) {
            throw RuntimeException("Critical dependency not satisfied. ")
        }
    }
}
```

## Investigation

When running `testDebugUnitTest` task, only `transformDebugUnitTestClassesWithAsm` is being executed, but not `transformDebugClassesWithAsm`. However both tasks are executed if Android Gradle Plugin is downgraded to 7.0.2

- Gradle tasks executed in `testDebugUnitTest`: https://pastebin.com/EAMxa3tu

## App for reproduce

https://github.com/jilulu/sample-agp71-hilt-app

**Comment by danysantiago** [comment]
Just wanted to give an update that a fix is being worked on the AGP side, thanks for filling a bug there too: https://issuetracker.google.com/213534628.

**Comment by danysantiago** [comment]
Based on the Google Issue:

> [fix] should be included in AGP 7.2 beta 2 and AGP 7.3 alpha 2. Unfortunately it's too late for this to be included in AGP 7.1 final, we're aiming for an early release of 7.1.1 which includes the fix.

**Comment by daespark** [comment]
Unsure if it's related, but I have bunch of test failing on 7.1.0 and 7.1.1 when using activityscenario with robolectric.

I have my vm initialized by  `val homeViewModel: HomeViewModel by viewModels()` in my fragment. I launch `HiltTestActivity` in a vm and bring up my frag, and it's unable to initialize vm correctly.

```
java.lang.RuntimeException: Cannot create an instance of class com.omada.prevent.home.HomeViewModel
	at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.java:269)
	at androidx.lifecycle.SavedStateViewModelFactory.create(SavedStateViewModelFactory.java:112)
	at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:185)
	at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:150)
	at androidx.lifecycle.ViewModelLazy.getValue(ViewModelProvider.kt:54)
	at androidx.lifecycle.ViewModelLazy.getValue(ViewModelProvider.kt:41)
	at com.omada.prevent.home.HomeFragment.getMHomeViewModel(HomeFragment.kt:93)
	at com.omada.prevent.home.HomeFragment.onCreate(HomeFragment.kt:107)

	```

Kotlin Version | 1.6.10
Dagger Hilt | 2.40.5
Robolectric | 4.6

Same tests work fine under AGP 7.1.0
